executor/distsql.go

```diff
@@ -652,29 +671,41 @@ func (w *indexWorker) fetchHandles(ctx context.Context, result distsql.SelectRes
 	}
 }
 
-func (w *indexWorker) extractTaskHandles(ctx context.Context, chk *chunk.Chunk, idxResult distsql.SelectResult) (handles []int64, err error) {
+func (w *indexWorker) extractTaskHandles(ctx context.Context, chk *chunk.Chunk, idxResult distsql.SelectResult) (
+	handles []int64, retChk *chunk.Chunk, err error) {
+	if w.isCheckOp {
+		chk = chunk.NewChunkWithCapacity(w.tps, w.maxChunkSize)
+	}
+	handleOffset := chk.NumCols() - 1
 	handles = make([]int64, 0, w.batchSize)
 	for len(handles) < w.batchSize {
 		chk.SetRequiredRows(w.batchSize-len(handles), w.maxChunkSize)
-		err = idxResult.Next(ctx, chk)
+		err = errors.Trace(idxResult.Next(ctx, chk))
 		if err != nil {
-			return handles, err
+			return handles, nil, err
 		}
 		if chk.NumRows() == 0 {
-			return handles, nil
+			return handles, retChk, nil
 		}
 		for i := 0; i < chk.NumRows(); i++ {
-			handles = append(handles, chk.GetRow(i).GetInt64(0))
+			h := chk.GetRow(i).GetInt64(handleOffset)
+			handles = append(handles, h)
+		}
+		if w.isCheckOp {
```

Comment:
This logic has nothing to do with this function, how about move this logic to line#662.
and change
```go
retChk = chunk.NewChunkWithCapacity(w.tps, w.maxChunkSize)
```
to 
```go
// no need to allocate max chunk size.
retChk = chunk.NewChunkWithCapacity(w.tps, chk.NumRows())
```